components/infobox/wikis/dota2/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,271 @@
+			Cell{name = 'Years Active (Coach)', content = {_args.years_active_coach}},
+			Cell{name = 'Years Active (Analyst)', content = {_args.years_active_analyst}},
+			Cell{name = 'Years Active (Talent)', content = {_args.years_active_talent}},
+		}
+	elseif id == 'history' then
+		if not String.isEmpty(_args.history_iwo) then
+			table.insert(widgets, Title{name = '[[Intel World Open|Intel World Open]] History'})
+			table.insert(widgets, Center{content = {_args.history_iwo}})
+		end
+		if not String.isEmpty(_args.history_gfinity) then
+			table.insert(widgets, Title{name = '[[Gfinity/Elite_Series|Gfinity Elite Series]] History'})
+			table.insert(widgets, Center{content = {_args.history_gfinity}})
+		end
+		if not String.isEmpty(_args.history_odl) then
+			table.insert(widgets, Title{name = '[[Oceania Draft League|Oceania Draft League]] History'})
+			table.insert(widgets, Center{content = {_args.history_odl}})
+		end
+	elseif id == 'role' then
+		return {
+			Cell{name = 'Current Role', content = {
+					CustomPlayer._createRole('role', _args.role),
+					CustomPlayer._createRole('role2', _args.role2)
+				}},
+		}
+	elseif id == 'nationality' then
+		return {
+			Cell{name = 'Location', content = {_args.location}},
+			Cell{name = 'Nationality', content = CustomPlayer._createLocations()}
+		}
+	end
+	return widgets
+end
+
+function CustomInjector:addCustomCells(widgets)
+	table.insert(widgets,
+		Builder{
+			builder = function()
+				local heroes = Player:getAllArgsForBase(_args, 'hero')
+				local icons = Array.map(heroes,
+					function(h, _)
+						return HeroIcon._getImage{hero = h, size = _SIZE_HERO}
+					end
+				)
+				return {
+					Cell{
+						name = 'Signature Hero',
+						content = {
+							table.concat(icons, '&nbsp;')
+						}
+					}
+				}
+			end
+		})
+	return widgets
+end
+
+function CustomPlayer:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomPlayer:makeAbbr(title, text)
+	if String.isEmpty(title) or String.isEmpty(text) then
+		return nil
+	end
+	return '<abbr title="' .. title .. '>' .. text .. '</abbr>'
+end
+
+function CustomPlayer:adjustLPDB(lpdbData)
+	lpdbData.status = lpdbData.status or 'Unknown'
+
+	lpdbData.extradata.role = _args.role
+	lpdbData.extradata.role2 = _args.role2
+	lpdbData.extradata.hero = _args.hero
+	lpdbData.extradata.hero2 = _args.hero2
+	lpdbData.extradata.hero3 = _args.hero3
+	lpdbData.extradata['lc_id'] = _base_page_name:lower()
+	lpdbData.extradata.team2 = mw.ext.TeamLiquidIntegration.resolve_redirect(
+		not String.isEmpty(_args.team2link) and _args.team2link or _args.team2 or '')
+
+	return lpdbData
+end
+
+function CustomPlayer:defineCustomPageVariables(args)
+	--retrieve birth month + day for storage in smw
+	local birthMonthAndDay = string.match(args.birth_date or '', '%-%d%d?%-%d%d?$')
+	birthMonthAndDay = string.gsub(birthMonthAndDay or '', '^%-', '')
+	Variables.varDefine('birthday', birthMonthAndDay)
```

Comment:
Template doesn't seem to be doing this. Seems to be the full birthday?
https://liquipedia.net/commons/index.php?title=Template:Birth_date_and_age&action=edit
https://liquipedia.net/dota2/index.php?title=Template:Infobox_player&action=edit#mw-ce-l422

Reply:
I checked with `{{#var:birthday}}` on the page and the values from old and new infobox match

Reply:
Isn't that because the Template/Dev is overwriting the var:birthday?
https://liquipedia.net/dota2/index.php?title=Template:Infobox_player/dev&action=edit#mw-ce-l77
Making these lines of code redundant?

Reply:
Ah. Got it.